src/algorithms/generate.rs

```diff
@@ -44,7 +42,10 @@ pub(crate) fn generate_multi_prime_key_with_exp<R: CryptoRngCore + ?Sized>(
         let prime_limit = (1u64 << (bit_size / nprimes) as u64) as f64;
 
         // pi aproximates the number of primes less than prime_limit
+        #[cfg(feature = "std")]
         let mut pi = prime_limit / (prime_limit.ln() - 1f64);
+        #[cfg(not(feature = "std"))]
+        let mut pi = prime_limit / (libm::logf(prime_limit as f32) as f64 - 1f64);
```

Comment:
this is not entirely correct or optimal, but I am not sure what other way there is

Reply:
@fjarri is there anything in `crypto-primes` for this?

Reply:
No, nothing like that currently